.github/workflows/rollout-module.yaml

```diff
@@ -0,0 +1,144 @@
+          [ -z "${{ vars.GIT_USER_NAME }}" ] && echo "GIT_USER_NAME is required" && exit 1 
+          [ -z "${{ vars.GH_TOOLS_HOST }}" ] && echo "GH_TOOLS_HOST is required" && exit 1 
+          [ -z "${{ vars.MODULE_MANIFESTS_REPO_NAME }}" ] && echo "MODULE_MANIFESTS_REPO_NAME is required" && exit 1 
+          echo "Validated"
+      - name: Use default (latest) Release Tag
+        if: inputs.releaseTag == 'use latest release'
+        shell: bash
+        run: |
+          latest=$(curl -s -H "Accept: application/vnd.github+json" ${ISTIO_API_URL}/releases/latest | jq -r '.tag_name') 
+          echo "TAG=${latest}" >> $GITHUB_ENV
+      - name: Validate given release tag
+        if: inputs.releaseTag != 'use latest release'
+        shell: bash
+        run: |
+          tags=$(curl -s -H "Accept: application/vnd.github+json" ${ISTIO_API_URL}/tags | jq -r '.[] | .name')
+          
+          if  echo $tags | tr " " '\n' | grep -F -q -x ${{ inputs.releaseTag }}; then
+              echo "TAG=${{ inputs.releaseTag }}" >> $GITHUB_ENV
+              echo "tag found"
+          else
+              echo "tag not found: ${{ inputs.releaseTag }}"
+              exit 1  
+          fi
+      - name: Set branch name
+        run: echo "BRANCH_NAME=istio-${TAG}-${{ inputs.channel }}" >> $GITHUB_ENV
+
+      - name: Setup git and clone repo
+        run: |
+          git config --global user.email ${{ vars.GIT_USER_EMAIL }}
+          git config --global user.name ${{ vars.GIT_USER_NAME }}
+          git clone ${MODULE_MANIFESTS_WITH_USER}
+        env:
+          GH_TOKEN: ${{ secrets.BOT_TOKEN_GITHUB_TOOLS }}
+
+      - name: Sync Repo and create branch
+        working-directory: module-manifests
+        run: |
+          git remote add upstream ${MODULE_MANIFESTS_WITH_USER}
+          git fetch upstream
+          git merge upstream/main
+          git checkout -B ${BRANCH_NAME}
+          mkdir -p modules/istio/${{ inputs.channel }}
+      - name: Download artifacts
+        working-directory: module-manifests/modules/istio/${{ inputs.channel }}
+        env:
+          CHANNEL: ${{ inputs.channel }}
+        run: |
+          if "${CHANNEL}" == "experimental"; then
+            curl -JL ${ISTIO_RELEASES_URL}/download/${TAG}/${EXPERIMENTAL_MANIFEST_FILENAME} >${MANIFEST_FILENAME}
+          else
+            curl -JL ${ISTIO_RELEASES_URL}/download/${TAG}/${MANIFEST_FILENAME} >${MANIFEST_FILENAME}
+          fi
+          curl -JL ${ISTIO_RELEASES_URL}/download/${TAG}/${DEFAULT_CR_FILENAME} >${DEFAULT_CR_FILENAME}
+      - name: Create module configuration
+        env:
+          CHANNEL: ${{ inputs.channel }}
+        working-directory: module-manifests/modules/istio/${{ inputs.channel }}
+        run: |
+          echo "Creating module configuration file:"
+          cat <<EOF | tee module-config.yaml
+          name: ${FULL_MODULE_NAME}
+          channel: ${CHANNEL}
+          version: ${TAG}
```

Comment:
For experimental channel we will need the version in the format of `${TAG}-${CHANNEL}` or hardcoded `${TAG}-experimental`